Document dynamic sel selectors (#82)
tscircuit/docs: docs/guides/using-sel-references.mdx

// selU2.doesnotexist // TypeScript error!
```

### Dynamic Reference Designators

> You can now call `sel` itself to create selectors for any reference designator.

Author: ```suggestion
You can call `sel` as a function to create selectors for any reference designator.
```

It's best to assign the resulting selector set to a variable so that you keep
type safety throughout your code:

```tsx
import { sel } from "tscircuit"
import { RP2040 } from "./rp2040"

const selU1 = sel.U1(RP2040)

> // Not recommended, but works

Author: ```suggestion
// More explicit version for custom scenarios 
```
